src/utils.rs

```diff
@@ -48,13 +42,13 @@ fn is_p2pkh(spk: &[u8]) -> bool {
     matches!(spk, [OP_DUP, OP_HASH160, OP_PUSHBYTES_20, .., OP_EQUALVERIFY, OP_CHECKSIG] if spk.len() == 25)
 }
 
-pub fn get_pubkey_from_input(vin: &VinData) -> Result<Option<PublicKey>, Error> {
-    if is_p2pkh(&vin.script_pub_key) {
-        match (&vin.txinwitness.is_empty(), &vin.script_sig.is_empty()) {
+pub fn get_pubkey_from_input(script_sig: &[u8], txinwitness: &Vec<Vec<u8>>, script_pub_key: &[u8]) -> Result<Option<PublicKey>, Error> {
+    if is_p2pkh(&script_pub_key) {
```

Comment:
I think we can remove the '&' here, also for the other else ifs

Reply:
Done